managedtenants/schemas/metadata.schema.yaml

```diff
@@ -129,6 +129,39 @@ properties:
       pattern: ^[A-Za-z0-9-_.]+$
     description: "Annotations to be applied to all objects created in the SelectorSyncSet."
   monitoring:
+    type: object
+    description: "[DEPRECATED] Configuration parameters to be injected in the ServiceMonitor used for federation. The target prometheus server found by matchLabels needs to serve service-ca signed TLS traffic (https://docs.openshift.com/container-platform/4.6/security/certificate_types_descriptions/service-ca-certificates.html), and it needs to be runing inside the monitoring.namespace, with the service name 'prometheus'."
+    additionalProperties: false
+    required:
+      - portName
+      - namespace
+      - matchNames
+      - matchLabels
+    properties:
+      portName:
+        type: string
+        pattern: ^[A-Za-z0-9][A-Za-z0-9-]{0,60}[A-Za-z0-9]$
+        description: "The name of the service port fronting the prometheus server."
+      namespace:
+        type: string
+        pattern: ^[A-Za-z0-9][A-Za-z0-9-]{0,60}[A-Za-z0-9]$
+        description: "Namespace where the prometheus server is running."
+      matchNames:
+        type: array
+        items:
+          type: string
+          format: printable
+          pattern: ^[a-zA-Z_:][a-zA-Z0-9_:]*$
+        description: "List of series names to federate from the prometheus server."
+      matchLabels:
+        type: object
+        items:
+          type: string
+          format: printable
+          pattern: ^[a-zA-Z_][a-zA-Z0-9_]*$
+        description: "List of labels used to discover the prometheus server(s) to be federated."
+        minItems: 1
+  metricsFederation:
```

Comment:
Why is this renamed to `metricsFederation`?

Reply:
previously metrics-federation config was provided under the `monitoring` section. And after realising that we need a new `monitoringStack` field, it was realised that `monitoring` and `monitoringStack` fields both in the addon metadata is pretty bad naming. Hence, the idea of renaming/deprecating the `monitoring` field to `metricsFederation` was formed.

That's why, if you notice, the description of the `monitoring` field begins with `[DEPRECATED]`

PS: the `monitoringStack` is totally unrelated to these `monitoring` and `metricsFederation` fields by the way